src/App.tsx

```diff
@@ -1,19 +1,167 @@
+/* eslint-disable max-len */
+/* eslint-disable no-console */
 /* eslint-disable jsx-a11y/control-has-associated-label */
-import React, { useContext, useEffect, useRef } from 'react';
+import classNames from 'classnames';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from 'react';
+import {
+  createTodo,
+  deleteTodo,
+  getTodos,
+  TodoData,
+  updateTodo,
+} from './api/todos';
 import { AuthContext } from './components/Auth/AuthContext';
+import { ProcessedContext } from './components/ProcessedContext/ProcessedContext';
+import { ErrorNotification } from './components/ErrorNotification/ErrorNotiication';
+import { NewTodoForm } from './components/NewTodoForm/NewTodoForm';
+import { TodoFilter } from './components/TodoFilter/TodoFilter';
+import { TodoList } from './components/TodoList/TodoList';
+import { ErrorTypes } from './types/ErrorTypes';
+import { FilterOptions } from './types/FilterOptions';
+import { Todo } from './types/Todo';
 
 export const App: React.FC = () => {
-  // eslint-disable-next-line @typescript-eslint/no-unused-vars
   const user = useContext(AuthContext);
-  const newTodoField = useRef<HTMLInputElement>(null);
+  const {
+    setError,
+    setIsAdding,
+    setProcessedTodoIds,
+  } = useContext(ProcessedContext);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filtredBy, setFiltredBy] = useState(FilterOptions.ALL);
+
+  const getUserTodos = async () => {
+    if (user) {
+      try {
+        const todosFromServer = await getTodos(user.id);
+
+        setTodos(todosFromServer);
+      } catch {
+        setError(ErrorTypes.GET);
+      }
+    }
+  };
+
+  const addNewTodo = async (todo: TodoData) => {
+    try {
+      setError(ErrorTypes.NONE);
+      setIsAdding(true);
+      const tempTodo = { ...todo, id: 0 };
+
+      setTodos(prevTodos => [...prevTodos, tempTodo]);
+
+      await createTodo(todo);
+    } catch {
+      setError(ErrorTypes.ADD);
+    } finally {
+      setIsAdding(false);
+      getUserTodos();
+    }
+  };
+
+  const deleteCurrentTodo = async (todoId: number) => {
+    try {
+      setError(ErrorTypes.NONE);
+      setProcessedTodoIds((currentIDs) => [...currentIDs, todoId]);
+
+      await deleteTodo(todoId);
+      setTodos(currentTodos => {
+        return currentTodos.filter(todo => todoId !== todo.id);
+      });
+    } catch {
+      setError(ErrorTypes.DELETE);
+    } finally {
+      setProcessedTodoIds([]);
+    }
+  };
+
+  const updateCurrentTodo = async (
+    todoId: number,
+    dataToUpdate: Partial<Todo>,
+  ) => {
+    try {
+      setError(ErrorTypes.NONE);
+      setProcessedTodoIds((currentIDs) => [...currentIDs, todoId]);
+      setTodos((currentTodos) => {
+        const updatedTodo = currentTodos.find(el => el.id === todoId) || null;
+
+        if (updatedTodo) {
+          Object.assign(updatedTodo, dataToUpdate);
+        }
+
+        return currentTodos;
+      });
```

Comment:
I dont know is it good or bad for optimization but for me code will look more simply if you will load them todos from server instead of setting todos on every update)